.github/workflows/main.yml

```diff
@@ -206,6 +206,8 @@ jobs:
           key: ${{ matrix.os }}-${{ matrix.feature }}
       - name: "root: cargo clippy"
         run: cargo clippy -F $FEATURE --all-targets
+      - name: "root: cargo clippy -F unstable"
+        run: cargo clippy -F $FEATURE -F unstable --all-targets
```

Comment:
I don't think we should have to run it with and without the "unstable" feature, probably just with it is good. Features are _supposed_ to be only additive (as in they shouldn't remove any code).
Making the CI job longer is also a non-goal.

Reply:
although this is now making me question why we run clippy both with and without the CUDA feature flag

Reply:
We have different code paths depending on feature flags. While it's true that they are additive from the workspace perspective, they still can cause exclusive sections of code to be enabled or disabled.

Reply:
I personally think this is a reasonable change for this specific issue.

Reply:
I also think we can just add it onto this existing clippy

Reply:
okay, I guess I'm wrong. You can totally have stuff compiled in only if the feature isn't enabled, and maybe that still fits within best practices, but I don't know how worried we are about that exactly